docs_src/design/adr/security/0019-spiffe.md

```diff
@@ -0,0 +1,338 @@
+   and uses it as a TLS client certificate to contact the
+   `security-spiffe-token-provider` service.
+   The EdgeX microservice uses the trust bundle as a server CA
+   to verify the TLS certificate of the remote service.
+1. `security-spiffe-token-provider` verifies the SVID using the trust bundle as client CA
+   to verify the client,
+   extracts the service key,
+   and issues an appropriate Vault service token.
+1. The EdgeX microservice accesses Vault as usual.
+
+
+#### Workload Registration and Agent Sockets
+
+The server uses a workload registration Unix domain socket that allows
+authorization entries to be added to the authorization database.
+This socket is protected by Unix file system permissions to control
+who is allowed to add entries to the database.
+
+In this proposal, a subcommand will be added to the EdgeX `secrets-config`
+utility to simplify the process of registering new services
+that uses the registration socket above.
+
+The agent uses a workload attesation Unix domain socket that
+is open to the world.  This socket is shared via a snap content-interface
+of via a shared host bind mount for Docker.  There is one agent per node.
+
+
+#### Trust Bundle
+
+SVID's must be traceable back to a known issuing authority (certificate authority)
+to determine their validity.
+
+In the proposed implementation, we will generate a CA on first boot and store it persistently.
+This root CA will be distributed as the trust bundle.
+The SPIFFE server will then generate a rotating sub-CA for issuing SVIDs,
+and the issued SVID will include both the leaf certificate and the intermediate certificate.
+
+
+#### Workload Authorization
+
+Workloads are authenticated by connecting to the `spiffe-agent`
+via a Unix domain socket, which is capable of identifying
+the process ID of the remote client.
+The process ID is fed into one of following workload attesters,
+which gather additional metadata about the caller:
+
+* The Unix workload attester gathers UID, GID, path, and SHA-256 hash of the executable.
+  The Unix workload attester would be used for plain services and snaps.
+* The Docker workload attester gathers container labels
+  that are added by docker-compose when the container is launched.
+  The Docker workload attester would be used for Docker-based EdgeX deployments.
+  An example label is `docker:label:com.docker.compose.service:edgex-core-data`
+  where the service label is the key value in the `services` section of the `docker-compose.yml`.
+  It is also possible to refer to labels built-in to the container image.
+* The Kubernetes workload attester gathers a wealth of pod and container metadata.
+
+Once authenticated, the metadata is sent to the SPIFFE server
+to authorize the workload.
+
+Workloads are authorized via an authorization database
+connected to the SPIFFE server.
+This database is configured with the known EdgeX services
+by the `spiffe-config` service,
```

Comment:
There's overlap here with our Service List ADR, and existing service registration...